app/src/main/java/fi/aalto/legroup/achso/entities/User.java

```diff
@@ -43,6 +45,24 @@ public void setUri(Uri uri) {
     }
 
     @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof User)) {
+            return false;
+        }
+        if (obj == this) {
+            return true;
+        }
+
+        User other = (User) obj;
+        return Objects.equal(name, other.name) && Objects.equal(uri, other.uri);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(name.hashCode(), uri.hashCode());
```

Comment:
Here you could use `Objects.hashCode(name, uri);`, right now it’s taking the hash code of a hash code.